R/database.R

```diff
@@ -623,488 +731,539 @@ get_combined_sv = function(min_vaf=0,
+  seg_df = data.frame(ID = unnested_df$region_segs$ID, CN = unnested_df$region_segs$CN,region_name = unnested_df$region_name)
   #arbitrarily take the first segment for each region/ID combination
-  seg_df = seg_df %>% dplyr::group_by(ID,region_name) %>% dplyr::slice(1) %>% dplyr::rename("sample_id"="ID")
+  seg_df = seg_df %>% 
+    dplyr::group_by(ID, region_name) %>% 
+    dplyr::slice(1) %>% 
+    dplyr::rename("sample_id" = "ID")
 
   #fill in any sample/region combinations with missing data as diploid
-  meta_arranged = get_gambl_metadata() %>% dplyr::select(sample_id,pathology,lymphgen) %>% arrange(pathology,lymphgen)
+  meta_arranged = get_gambl_metadata() %>% 
+    dplyr::select(sample_id, pathology, lymphgen) %>% 
+    arrange(pathology, lymphgen)
 
-  eg = expand_grid(sample_id=pull(meta_arranged,sample_id),region_name=as.character(unique(seg_df$region_name)))
-  all_cn = left_join(eg,seg_df,by=c("sample_id"="sample_id","region_name"="region_name")) %>%
-    mutate(CN=replace_na(CN,2))
-  cn_matrix = pivot_wider(all_cn,id_cols="sample_id",names_from="region_name",values_from = "CN") %>%
+  eg = expand_grid(sample_id = pull(meta_arranged, sample_id), region_name = as.character(unique(seg_df$region_name)))
+  all_cn = left_join(eg, seg_df, by = c("sample_id" = "sample_id", "region_name" = "region_name")) %>%
+    mutate(CN = replace_na(CN, 2))
+
+  cn_matrix = pivot_wider(all_cn, id_cols = "sample_id", names_from = "region_name", values_from = "CN") %>%
     column_to_rownames("sample_id")
+
   #order the regions the same way the user provided them for convenience
   return(cn_matrix[,region_names])
 }
 
-#' Get all segments for a single sample_id
+
+#' Get all segments for a single sample_id.
 #'
-#' @param sample_id The sample_id for the sample to retrieve segments for
-#' @param with_chr_prefix Set to TRUE to add a chr prefix to chromosome names
-#' @param streamlined Return a minimal output rather than full details
+#' @param this_sample_id The sample_id for the sample to retrieve segments for.
+#' @param with_chr_prefix Set to TRUE to add a chr prefix to chromosome names.
+#' @param streamlined Return a minimal output rather than full details.
+#' @param from_flatfile Resort to a flat file instead of the database for retrieving the data.
 #'
-#' @return
+#' @return A list of segments for a specific sample ID
 #' @export
 #'
 #' @examples
-get_sample_cn_segments = function(this_sample_id,with_chr_prefix=FALSE,streamlined=FALSE,from_flatfile=FALSE){
-  sample_status = get_gambl_metadata() %>%
-    dplyr::filter(sample_id==this_sample_id) %>%
+#' sample_cn_seg = get_sample_cn_segments("some-sample-id", FALSE, FALSE, FALSE)
+#'
+get_sample_cn_segments = function(this_sample_id, 
+                                  with_chr_prefix = FALSE, 
+                                  streamlined = FALSE, 
+                                  from_flatfile = FALSE){
+
+  sample_status = get_gambl_metadata() %>% 
```

Comment:
I've been running into issues with this function. Could you update it to take `gambl_md` as an argument and use that in place of `get_gambl_metadata()`? Also I don't think it runs if `from_flatfile == TRUE` because it requires a `genome_build` value for each sample and that is not currently generated.

Reply:
I am not sure if this is what you meant, but one solution could be to add a new parameter that returns the output from `get_gambl_metadata` as a variable (df) that can be used to pull samples from instead of calling get_gambl_metadata every time if this "new" parameter is set to TRUE? Similarly to what I did for [this](https://github.com/morinlab/GAMBLR/issues/50)

Reply:
Support to return cn segments for multiple samples have been added in [this](https://github.com/morinlab/GAMBLR/commit/846977bf7cc53c74a90df6e49d334c33e87242b3) commit. 

This was done by adding two new parameters `multiple_samples` and `sample_list`. If `multiple_samples` is set TRUE (default is FALSE), the function expects a list of samples (samples_list). 

The updated function was successfully tested with one sample, multiple samples as a list (vector of characters), and with a csv-file (one sample per row). Examples were also updated together with function-description.